src/dil_demo/i18n.clj

```diff
@@ -0,0 +1,93 @@
+;;; SPDX-FileCopyrightText: 2024 Jomco B.V.
+;;; SPDX-FileCopyrightText: 2024 Topsector Logistiek
+;;; SPDX-FileContributor: Joost Diepenmaat <joost@jomco.nl>
+;;; SPDX-FileContributor: Remco van 't Veer <remco@jomco.nl>
+;;;
+;;; SPDX-License-Identifier: AGPL-3.0-or-later
+
+(ns dil-demo.i18n
+  (:require [clojure.java.io :as io]
+            [clojure.set :as set]
+            [clojure.string :as string]
+            [clj-yaml.core :as yaml])
+  (:import (java.time ZonedDateTime)
+           (java.time.temporal ChronoUnit)))
+
+(def ^:dynamic lang "nl")
+
+(defn- check-translations
+  "Throw an exception when languages in `translations` (top level keys)
+  do not have the same keys."
+  [translations]
+  (let [langs (keys translations)
+        all-keys (into #{} (->> translations
+                                (map second)
+                                (mapcat keys)
+                                (set)))
+        missing (reduce (fn [m lang]
+                          (let [d (set/difference all-keys
+                                                  (-> translations (get lang) keys))]
+                            (if (seq d)
+                              (assoc m lang d)
+                              m)))
+                        {}
+                        langs)]
+    (when (seq missing)
+      (throw (ex-info "missing translations" missing)))
+
+    translations))
+
+(defn- read-translations []
+  (with-open [rd (io/reader (io/resource "i18n.yml"))]
+    (yaml/parse-stream rd :keywords false)))
+
+(def translations
+  (delay (-> (read-translations)
+             (check-translations))))
+
+(def languages (delay (set (keys @translations))))
+
+(defn has-language? [^String lang]
+  (contains? (into #{} (map name @languages)) lang))
+
+(defn cookie-expires []
+  (.plus (ZonedDateTime/now) 1 ChronoUnit/YEARS))
+
+(def ^:dynamic throws-exception true)
+
+(defn wrap [app & {:keys [throw-exceptions] :or {throw-exceptions true}}]
```

Comment:
als we toch al een `wrap` nodig hebben voor configuratie, is een global `@languages` en `@translations` delay wel nodig?

Ik verwacht daar issues mee als we ooit meer dan 1 app tegelijk willen starten in 1 runtime